lib/fluent/plugin/parser_auditd.rb

```diff
@@ -14,7 +14,7 @@ def configure(conf={})
     def parse(text)
       begin
         parsed_line = @auditd.parse_auditd_line text
-        time = parsed_line.nil? ? nil : parsed_line['time']
+        time = parsed_line.nil? ? nil : DateTime.parse(parsed_line['time']).to_time.to_f
```

Comment:
The original value is a string like this: `"1506434946.557"` so you can just use `.to_f` on that string:
```
begin
  time = parsed_line['time'].to_f
rescue
  time = 0.0
end
time = gettimefromfluentd if time.eql?(0.0)
```
Fluentd usually generates an internal time stamp - the `time` parameter to many emit and filter methods - to assign to records that have no other time.  Not sure where you get that in this context, or maybe you can return `nil` for `time` here and fluentd will use its internal time?

Reply:
`parsed_line['time']` is in rfc3339 format as type of string. But fluentd requires numeric type.

```
irb(main):065:0> 'foo'.to_f
=> 0.0
```
Not sure if it throws exception. I'll try to send nil and see if fluentd generates timestamps. Nevertheless, the message already contains a timesptamp

Reply:
Ah, I see - `parsed_line['time']` is already in the rfc3339 string format

Then, rather than having to parse the time back from a string, you could just do something like this:
```
def normalize(target)
       event[TIME]                                     = target[TIME].to_f
....

     def parse(text)
       begin
         parsed_line = @auditd.parse_auditd_line text 
         unless parsed_line.nil?
           time = parsed_line[TIME]
           parsed_line[TIME] = Time.at(parsed_line[TIME]).utc.to_datetime.rfc3339(6)
...
```